netbox/extras/models/customfields.py

```diff
@@ -627,3 +631,46 @@ def validate(self, value):
 
         elif self.required:
             raise ValidationError("Required field cannot be empty.")
+
+
+class CustomFieldChoiceSet(ChangeLoggedModel):
+    """
+    Represents a set of choices available for choice and multi-choice custom fields.
+    """
+    name = models.CharField(
+        max_length=100,
+        unique=True
+    )
+    description = models.CharField(
+        max_length=200,
+        blank=True
+    )
+    extra_choices = ArrayField(
+        base_field=models.CharField(max_length=100),
```

Comment:
Not to change here but in relation to the issue of increasing field sizes - these max length are a bit of a magic number in that the choices and selection fields must match and are 100, where description are 200 - seems like these should be constants in the code. Just mentioning as I'm thinking of it, can open a new issue if you agree.

Reply:
No need to change the field size; we're simply porting what has existed for years.